src/main/java/frc/robot/subsystems/vision/readme.md

```diff
@@ -0,0 +1,41 @@
+# Vision Subsystem
+
+Provides a stateless interface VisionProvider for access to functionality.
+Decorator interfaces handle specific functionality such as invalidating cached values.
+Should be functionally similar and compatible with 2020's VisionSystem without the 
+statefulness and weak points
+
+# The doctrine known as EOS-P
+
+EOS-P stands for Elegant Objects - SolarMC Combined Programming Paradigm.
+It is a style of code that incorporates everything i've learned from every website and project i've worked on. 
+It is in no way complete and in no way 100% correct, but it is my preference.
+I do not want to betray the constitution's programming paradigm or FRC codestyle (I'll call it FRC-P)
+but I would rather use EOS-P as it avoids bugs and timewasting "magic" code
+by design.
+
+**Some interesting rules of EOS-P**
+
+1. A Minimum Viable Project is more important than maintainability.
+2. Maintainability is more important than performance.
+3. Performance is more important than simplicity.
```

Comment:
wut

Reply:
Let me elaborate on this in my next PR! Please explain what you are confused by.